com.unity.cinemachine/Runtime/Debug/CinemachineDebug.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using System.Text;
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEngine.UIElements;
+#endif
+
+namespace Unity.Cinemachine
+{
+    /// <summary>Manages onscreen positions for Cinemachine debugging output</summary>
+    static class CinemachineDebug
+    {
+        static List<StringBuilder> s_AvailableStringBuilders;
+
+#if CINEMACHINE_UNITY_IMGUI
+        /// <summary>Reserve an on-screen rectangle for debugging output.</summary>
+        /// <param name="client">The client caller.  This is used as a handle.</param>
+        /// <param name="text">Sample text, for determining rectangle size</param>
+        /// <param name="style">What style will be used to draw, used here for
+        /// determining rect size</param>
+        /// <returns>An area on the game screen large enough to print the text
+        /// in the style indicated</returns>
+        public static Rect GetScreenPos(Camera camera, string text, GUIStyle style)
+        {
+            var size = style.CalcSize(new GUIContent(text));
+            var pos = Vector2.zero;
+            if (camera != null)
+            {
+                var r = camera.rect; 
+                pos.x += r.x * Screen.width;
+                pos.y += (1 - r.y - r.height) * Screen.height;
+            }
+            return new Rect(pos, size);
+        }
+#endif
+
+        
+#if UNITY_EDITOR
+        const string k_DebugUIName = "CinemachineDebugUI";
+        static Dictionary<Camera, VisualElement> s_CameraViewRectContainers = new();
+        static GameObject s_UIDocumentHolder;
+        static UIDocument s_UIDocument;
+        public static VisualElement CreateRuntimeUIContainer(Camera outputCamera)
+        {
+            if (s_UIDocumentHolder == null)
+            {
+                s_UIDocumentHolder = GameObject.Find(k_DebugUIName);
+                if (s_UIDocumentHolder == null)
+                {
+                    s_UIDocumentHolder = new GameObject(k_DebugUIName)
+                    {
+                        hideFlags = HideFlags.NotEditable | HideFlags.HideInHierarchy,
+                        tag = "EditorOnly" // TODO: we may want runtime too
+                    };
+                }
+            }
+            if (!s_UIDocumentHolder.TryGetComponent(out s_UIDocument))
+            {
+                s_UIDocument = s_UIDocumentHolder.AddComponent<UIDocument>();
+                
+                const string path = "Packages/com.unity.cinemachine/Runtime/UI/";
```

Comment:
Should this asset be in the Debug directory?

Reply:
Yes!